### PR TITLE
MM-35355 Detox/E2E: Add e2e for channels settings

### PR DESCRIPTION
**Pull request by josephbaylon**
#### Summary
- Added tests for MM-T850, MM-T851, MM-T853, MM-T854, MM-T878, MM-T3317 in
    - `detox/e2e/test/channel_moderation/remove_user_from_channel.e2e.js` 
    - `detox/e2e/test/channel_settings/channel_add_members.e2e.js`
    - `detox/e2e/test/channel_settings/channel_view_members.e2e.js`
    - `detox/e2e/test/channels/favorite_channels.e2e.js`
    - `detox/e2e/test/channels/pinned_messages.e2e.js`
- Cleaned up doc links and sorted functions in api files
- Fixed `apiGetChannelByName` implementation and added `apiGetChannelByNameAndTeamName`

Note: The important changes are the list of tests above. The rest are cleanup.

#### Ticket Link
JIRA tickets:
- https://mattermost.atlassian.net/browse/MM-35355

Zephyr cases:
- [MM-T850](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T850)
- [MM-T851](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T851)
- [MM-T853](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T853)
- [MM-T854](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T854)
- [MM-T878](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T878)
- [MM-T3317](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3317)

#### Screenshots
Screen Shot 2021-05-07 at 1 13 26 PM
Screen Shot 2021-05-07 at 10 13 16 AM
Screen Shot 2021-05-07 at 10 16 14 AM
Screen Shot 2021-05-07 at 10 17 34 AM
Screen Shot 2021-05-07 at 11 31 10 AM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: detox/e2e/support/server_api/channel.js [hunk 1]
```diff
@@ -18,9 +18,29 @@ import {getResponseFromError} from './common';
...
+        return {member: response.data};
+    } catch (err) {
+        return getResponseFromError(err);
+    }
+};
+
 /**
  * Create a channel.
- * See https://api.mattermost.com/#tag/channels/paths/~1channels/post
+ * See https://api.mattermost.com/#operation/CreateChannel
```
Comment (saturninoabril) [inline detox/e2e/support/server_api/channel.js:43]: I see. We'll need to update the references in Webapp/Cypress as well.

File: detox/e2e/support/server_api/plugin.js [hunk 2]
```diff
@@ -18,41 +18,84 @@ import {apiUploadFile, getResponseFromError} from './common';
...
+];
+
 /**
- * Get plugins.
- * See https://api.mattermost.com/#tag/plugins/paths/~1plugins/get
- * @return {Object} returns {plugins} on success or {error, status} on error
+ * Disable non-prepackaged plugins.
  */
-export const apiGetAllPlugins = async () => {
+export const apiDisableNonPrepackagedPlugins = async () => {
```
Comment (saturninoabril) [inline detox/e2e/support/server_api/plugin.js:40]: I don't see this being used right now but it should be called at `init.js` / `beforeAll`.

**Review by saturninoabril — APPROVED** [review]
Thanks Joseph! Tested and passed.

File: detox/e2e/support/server_api/plugin.js (hunk 2 above)
Comment (josephbaylon) [reply detox/e2e/support/server_api/plugin.js:40]: Added and tested

**Review by enahum — APPROVED** [review]
LGTM

**Pull request merged**